core/corehttp/logs.go

```diff
@@ -41,6 +42,7 @@ func (w *writeErrNotifier) Close() error {
 	case w.errs <- io.EOF:
 	default:
 	}
+	close(w.errs)
```

Comment:
i'm concerned this might get called twice and panic, want to maybe nil it out after closing, and do the other appropriate checks on the nil value?